Libs/MRML/Widgets/qMRMLSliceControllerWidget.cxx

```diff
@@ -1997,6 +2007,40 @@ void qMRMLSliceControllerWidget::fitSliceToBackground()
   d->SliceLogic->EndSliceNodeInteraction();
 }
 
+void qMRMLSliceControllerWidget::MaximizeLayout() {
+  Q_D(qMRMLSliceControllerWidget);
+  const char* MaxLayoutName = d->MRMLSliceNode->GetLayoutName();
+  vtkMRMLLayoutNode* layoutNode = vtkMRMLLayoutNode::SafeDownCast(this->mrmlScene()->GetSingletonNode("vtkMRMLLayoutNode", "vtkMRMLLayoutNode"));
+  if(QString::fromUtf8(MaxLayoutName).contains("Red")) {
+    if(!this->maximumView) {
+      layoutNode->SetViewArrangement(vtkMRMLLayoutNode::SlicerLayoutOneUpRedSliceView);
+      this->maximumView = true;
+    }
+    else{
+      layoutNode->SetViewArrangement(vtkMRMLLayoutNode::SlicerLayoutFourUpView);
+      this->maximumView = false;
+    }
```

Comment:
> I think it would be better to have the View node keep track of the layout that was active before maximizing one view and then swap back to that layout when minimizing.

I agree with @pieper here. If I'm in "Side by side" layout (Red on left, Yellow on right), then when I maximize the Red view (Red Only) and then minimize it again, I would expect it go to my previous layout of "Side by side" and not "Four-Up".